src/manipulateToDo.js

```diff
@@ -0,0 +1,88 @@
+let dragStore = null;
+const onDragOver = (ev) => {
+  ev.preventDefault();
+  const elUnder = ev.target;
+  const pElement = elUnder.parentElement;
+  const parentName = pElement.nodeName;
+  if (parentName === 'DIV') {
+    const actualDivClass = ev.target.classList[0];
+    if (actualDivClass === 'to-do-row') {
+      elUnder.parentElement.insertBefore(dragStore, elUnder);
+    } else if (actualDivClass === 'to-do') {
+      elUnder.parentElement.parentElement.parentElement.insertBefore(
+        dragStore,
+        elUnder.parentElement.parentElement,
+      );
+    } else if (actualDivClass === 'fas') {
+      elUnder.parentElement.parentElement.insertBefore(
+        dragStore,
+        elUnder.parentElement,
+      );
+    } else if (actualDivClass === 'checkbox') {
+      elUnder.parentElement.parentElement.parentElement.insertBefore(
+        dragStore,
+        elUnder.parentElement.parentElement,
+      );
+    }
+  }
+};
+
+const dragEnd = () => {
+  setTimeout(() => {
+    dragStore.style.backgroundColor = '#fff';
+  }, 1000);
+};
+
+const drag = (ev) => {
+  const draggedElement = ev.target;
+  dragStore = draggedElement;
+  dragStore.style.backgroundColor = '#4c3c3c';
+};
+
+const dragStart = (e) => {
+  e.target.style.backgroundColor = '#4c3c3c';
+};
+
+const removeOne = (task, tasks, showToDo) => {
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  tasks.splice(tasks.indexOf(task), 1);
+  // eslint-disable-next-line no-use-before-define
+  showToDo(tasks);
+};
+
+const removeSelected = (tasks, showToDo, goToInput) => {
+  tasks = tasks.filter((task) => !task.completed);
+  localStorage.setItem('tasks', JSON.stringify(tasks));
+  showToDo(tasks);
+  goToInput();
+  return true;
+};
+
+const addToDo = (tasks, showToDo, goToInput) => {
+  const description = document.querySelector('.input').value;
+  if (!description) {
+    goToInput();
+    return false;
+  }
+  const indexArray = tasks.map(({ index }) => index);
+  const sortedIndeces = indexArray.sort((a, b) => a - b);
+  let highestIndex = sortedIndeces[sortedIndeces.length - 1];
+  if (Number.isNaN(highestIndex) || !highestIndex) {
+    highestIndex = 0;
+  }
```

Comment:
- [ ] After deleting the selected task on clicking `clear all completed`, It works fine to delete them. But when I add a new task the deleted task comes up with a new item. This is strange behaviour with localStorage. Kindly try to fix this issue.

Initial List
![image](https://user-images.githubusercontent.com/53019916/130089380-2a1a7da0-585d-4b95-a9db-62e1446f6505.png)

After deleting the Selected item on clicking `Clear all Completed`
![image](https://user-images.githubusercontent.com/53019916/130089568-e452c765-b1d4-4015-86aa-350180fc6838.png)

After adding again a new task
![image](https://user-images.githubusercontent.com/53019916/130089700-2601f878-08da-48d6-9416-e25e5f0a7c20.png)

Reply:
Thank you so much, @hemant-soni-vst-au4:pray: 
I have done my best to get the bug fixed. I am heading over to my dashboard to resubmit another request.
I am grateful:raised_hands: